File: app/src/main/java/com/dialogapp/dialog/db/AccountDao.java
package com.dialogapp.dialog.db;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;

import com.dialogapp.dialog.model.AccountResponse;

@Dao
public interface AccountDao {
    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void insert(AccountResponse accountResponse);

    @Query("SELECT * FROM account WHERE username = :username")
    LiveData<AccountResponse> fetchAccountInfo(String username);

    @Query("DELETE FROM account")
    void dropAccount();
}

File: app/src/main/java/com/dialogapp/dialog/db/AccountDao.kt
package com.dialogapp.dialog.db

import androidx.room.Dao
import androidx.room.Insert
import androidx.room.OnConflictStrategy
import androidx.room.Query
import com.dialogapp.dialog.model.LoggedInUser

@Dao
interface AccountDao {
    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insert(loggedInUser: LoggedInUser)

    @Query("DELETE FROM logged_in_user")
    fun deleteAccount()

    @Query("SELECT * FROM logged_in_user")
    fun loadAccount(): LoggedInUser?
}
